packages/dev/sharedUiComponents/src/lines/optionsLineComponent.tsx

```diff
@@ -44,20 +48,32 @@ export class OptionsLineComponent extends React.Component<IOptionsLineComponentP
     constructor(props: IOptionsLineComponentProps) {
         super(props);
 
-        this.state = { value: this._remapValueIn(this._getValue(props)) };
+        this.state = { value: this._remapValueIn(this._getValue(props)), addCustom: false };
     }
 
-    shouldComponentUpdate(nextProps: IOptionsLineComponentProps, nextState: { value: number }) {
+    shouldComponentUpdate(nextProps: IOptionsLineComponentProps, nextState: { value: number; addCustom: boolean }) {
         if (this._localChange) {
             this._localChange = false;
+
             return true;
         }
 
         const newValue = this._remapValueIn(nextProps.extractValue ? nextProps.extractValue(this.props.target) : nextProps.target[nextProps.propertyName]);
+
         if (newValue != null && newValue !== nextState.value) {
             nextState.value = newValue;
+
+            return true;
+        }
+
+        if (this.props.options !== nextProps.options) {
+            return true;
+        }
+
+        if (this.props.options !== nextProps.options) {

```

Comment:
Isn't this a repeat of the above condition?

Reply:
It is! That must have happened while solving the merge conflicts. I've removed it!